File: samples/CslaNet/cs/DeepData/DeepData.Library/LineItemDetailList.cs
using Csla;
using Csla.Data;

namespace DeepData.Library
{
	[global::System.Serializable()]
	public partial class LineItemDetailList
	  : ReadOnlyListBase<LineItemDetailList, LineItemDetailInfo>
	{

	#region " Factory Methods "

	  internal static LineItemDetailList NewList()
	  {
	  	return new LineItemDetailList();

	  }

		private LineItemDetailList(){}

	#endregion

	#region " Data Access "

	  internal void LoadChild(SafeDataReader data)
	  {
	  	IsReadOnly = false;
	  	Add(LineItemDetailInfo.GetItem(data));
	  	IsReadOnly = true;
	  }

	#endregion

	}
}

File: Samples/CslaNet/vb/DeepData/DeepData.Library/LineItemDetailList.vb
<Serializable()> _
Public Class LineItemDetailList
  Inherits ReadOnlyListBase(Of LineItemDetailList, LineItemDetailInfo)

#Region " Factory Methods "

  Friend Shared Function NewList() As LineItemDetailList

    Return New LineItemDetailList

  End Function

  Private Sub New()

  End Sub

#End Region

#Region " Data Access "

  Friend Sub LoadChild(ByVal data As SafeDataReader)

    IsReadOnly = False
    Add(LineItemDetailInfo.GetItem(data))
    IsReadOnly = True

  End Sub

#End Region

End Class
